src/v/datalake/translation/state_machine.h

```diff
@@ -103,7 +103,7 @@ class translation_stm final : public raft::persisted_stm<> {
 
     void update_highest_translated_offset(kafka::offset new_offset);
 
-    kafka::offset _highest_translated_offset{};
+    std::optional<kafka::offset> _highest_translated_offset;
```

Comment:
Can you add a comment describing the distinction between nullopt and kafka::offset{}? Or is this never kafka::offset{}?

Reply:
I simplified this as it doesn't makes sense to have optional and not initialised as they have the same semantics